### PR TITLE
Refactor: Create Safe Status hooks

### LINKED ISSUE #932: [Safe Creation] Status screen
## What is the feature about

At the end of the safe creation flow, the user should see a number of status updates depending on the state of the transaction.

## The list of requirements

- Display messages and actions for [every possible status](https://www.notion.so/gnosis-safe/Revamp-bc8783e2eaf041d0b6a50c9a1641aee7), implement a state machine pattern for this
- Display pre-computed safe address and transaction hash when possible
- Navigate user to their dashboard and display a modal with instructions/tips
- (optional) Add a nice animation for the safe logo (e.g. something like https://drive.google.com/file/d/1MdKjV0g0NwXYXE_eH6NEZRl33eoNf_CW/view?usp=sharing but infinite/loading)
- Make it work for dark mode

## Designs/sketches
Screenshot 2022-10-18 at 16 10 50
Screenshot 2022-10-18 at 16 11 17
Screenshot 2022-10-18 at 16 11 29

## Links

Figma: https://www.figma.com/file/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?node-id=56%3A7045

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #932 

Redesign will be done in a separate PR

## How this PR fixes it

- Refactors the folder structure for create safe status related components and hooks
- Removes `AWAITING_WALLET` state and instead disables the retry button
- Simplifies `useSafeCreation` hook and adds more tests
- Moves `CreationStatus` component into the stepper
- Removes `usePendingSafe` hook
- Renames `useWatchSafeCreation` to `useSafeCreationEffects`

## ToDos

- [x] Implement test cases
- [x] Visually hide stepper when in status step

## How to test it

The following messages should be visible to the user during safe creation: https://www.notion.so/gnosis-safe/Revamp-bc8783e2eaf041d0b6a50c9a1641aee7

1. Create safe, confirm in wallet -> AWAITING, PROCESSING, SUCCESS
2. Create safe, reject in wallet -> AWAITING, WALLET_REJECTED
3. Create safe, confirm in wallet, reload the page -> PROCESSING, SUCCESS
4. Create safe, reject in wallet, reload the page -> AWAITING
5. Create safe, reject in wallet, disconnect wallet -> Retry button should be disabled and show a Tooltip on hover
6. Create safe, confirm in wallet, speed-up tx -> AWAITING, PROCESSING, SUCCESS
7. Create safe, confirm in wallet, cancel tx -> AWAITING, PROCESSING, ERROR
8. Create safe, confirm in wallet, reload page, speed-up tx -> PROCESSING, SUCCESS
9. Create safe, confirm in wallet, reload page, cancel tx -> PROCESSING, ERROR
10. Create safe, confirm in wallet, speed-up tx, reload page -> PROCESSING, SUCCESS
11. Create safe, confirm in wallet, cancel tx, reload page -> PROCESSING, ERROR

### CODE REVIEW

**Review by katspaugh — APPROVED** [review]
🚀

**Comment by DiogoSoaress** [comment]
Is the white color in the green background per design? I would mind it before merging as it really hurt the eyes
Screenshot 2022-10-26 at 17 43 47

File: src/components/create-safe/status/CreationStatus.tsx [hunk 1]
```diff
@@ -1,154 +1,113 @@
...
+  onSubmit: StepRenderProps['onSubmit']
+  onBack: StepRenderProps['onBack']
+  setStep: StepRenderProps['setStep']
 }
 
-const getStep = (status: SafeCreationStatus) => {
-  const loading = (
-    <img src="/images/open/safe-creation-process.gif" alt="Image of a vault that is loading" className={css.loading} />
+export const CreationStatus = ({ params, setStep }: Props) => {
+  const [status, setStatus] = useStatus()
```
Comment (DiogoSoaress) [inline src/components/create-safe/status/CreationStatus.tsx:31]:
```suggestion
  const [status, setStatus] = useState<SafeCreationStatus>(SafeCreationStatus.AWAITING)
```

the `useStatus` hook is unnecessary

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]

**Comment by JagoFigueroa** [comment]
Good morning! Testing this scenario:

- Create safe, reject in wallet, reload the page -> AWAITING

It is redirecting me to the first step of safe creation. Is that expected?

https://user-images.githubusercontent.com/92332710/198208492-4121280f-fc7b-4c17-baf8-e1e3f8bf7548.mp4

Cheers

**Comment by katspaugh** [comment]
> Is the white color in the green background per design? I would mind it before merging as it really hurt the eyes
Out of scope, there's a separate issue for that.

**Comment by JagoFigueroa** [comment]
For this scenario: 

* Create safe, confirm in wallet, speed-up tx, reload page -> PROCESSING, SUCCESS

I am seeing a weird animation in the UI, in case you wanna have a look 😄  

https://user-images.githubusercontent.com/92332710/198215096-da285a4f-aea8-4a44-a402-3f3ed0fb808c.mp4

File: src/components/create-safe/status/CreationStatus.tsx (hunk 1 above)
Comment (usame-algan) [reply src/components/create-safe/status/CreationStatus.tsx:31]: I added it to make testing the component easier. Alternatively, we could also mock react useState.

**Comment by katspaugh** [comment]
If you slow down the video to 0.5x, it's basically just the loading GIF jumping to a bigger size just before changing to the next image and going to the next status.
Probably the same image element is being re-rendered and the size is immediate whereas the `src` attribute` isn't.
Should be easy to fix by a key prop to the img tag with the same value as the src.

File: src/components/create-safe/status/CreationStatus.tsx (hunk 1 above)
Comment (DiogoSoaress) [reply src/components/create-safe/status/CreationStatus.tsx:31]: I think we should create a new hook if not needed. Should be possible to mock useState

**Comment by usame-algan** [comment]
> Good morning! Testing this scenario:
> Create safe, reject in wallet, reload the page -> AWAITING
> It is redirecting me to the first step of safe creation. Is that expected?

Thanks for testing! I changed it so that it will stay on the status page even when rejecting for the first time.

**Comment by usame-algan** [comment]
> I am seeing a weird animation in the UI, in case you wanna have a look 😄

Added the explicit width + height to the other images as well and it stopped jumping for me. Can you check again?

**Comment by usame-algan** [comment]
✅ Tested all cases

**Comment by JagoFigueroa** [comment]
All cases are working fine for me as well on Goerli & Gnosis Chain after the changes.

I will talk with señor Franco to give this ticket a final check when he is back from vacation as I have to end my week early, sorry.

**Pull request merged**